From 33453b930ba918b92e9bf6666d52834cd622a38e Mon Sep 17 00:00:00 2001
From: Nithin Dabilpuram <ndabilpuram@marvell.com>
Date: Fri, 28 Jun 2019 15:59:07 +0530
Subject: [PATCH 315/386] octeontx2-af: Fix inner l4 layer id

Current NPC profile has designated Inner L4 layer as LH. This fixes
RSS flow key alg preperation to use LH instead of LG.

Change-Id: If14dad7ffa1f1300bd6f412394e59fd451610f74
Signed-off-by: Nithin Dabilpuram <ndabilpuram@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/11759
Reviewed-by: Naveen Mamindlapalli <Naveen.Mamindlapalli@cavium.com>
Reviewed-by: Kiran Kumar Kokkilagadda <Kiran.Kokkilagadda@cavium.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
index 20c9af7aa33d..caeb959f7d67 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
@@ -2625,7 +2625,7 @@ static int set_flowkey_fields(struct nix_rx_flowkey_alg *alg, u32 flow_cfg)
 			if (key_type == NIX_FLOW_KEY_TYPE_INNR_TCP ||
 			    key_type == NIX_FLOW_KEY_TYPE_INNR_UDP ||
 			    key_type == NIX_FLOW_KEY_TYPE_INNR_SCTP)
-				field->lid = NPC_LID_LG;
+				field->lid = NPC_LID_LH;
 			field->bytesm1 = 3; /* Sport + Dport, 4 bytes */
 
 			/* Enum values for NPC_LID_LD and NPC_LID_LG are same,
-- 
2.17.1

